apps/sms/views/conversation/js/attachment.js

```diff
@@ -26,7 +26,8 @@
   function Attachment(blob, options) {
     options = options || {};
     this.blob = blob;
-    this.name = blob.name || options.name || '';
+    this.name = typeof blob.name == 'string' ? blob.name : (
+      typeof options.name == 'string' ? options.name : '');
```

Comment:
maybe it's more readable to keep the previous line, and add a new one `if (typeof this.name !== 'string') { this.name = '' }`

What do you think ?